src/tempo.py

```diff
@@ -104,6 +105,24 @@ def url(self) -> str:
         scheme = "https" if self.tls_ready else "http"
         return f"{scheme}://{self._external_hostname}"
 
+    def get_receiver_url(self, protocol: ReceiverProtocol, ingress: TraefikRouteRequirer):
+        """Return the receiver endpoint URL based on the protocol.
+
+        if ingress is used, return endpoint provided by the ingress instead.
+        """
+        protocol_type = ReceiverProtocolType.get(protocol)
+        is_ingress = ingress.is_ready()
```

Comment:
nit: 

```suggestion
        has_ingress = ingress.is_ready()
```

or `ingress_is_ready`